x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline_button.test.tsx

```diff
@@ -6,69 +6,118 @@
+
+const props = {
+  initialFocus: 'title' as const,
+  timelineId: 'timeline-1',
+  toolTip: 'tooltip message',
+};
+
+const TestSaveTimelineButton = (_props: SaveTimelineComponentProps) => (
+  <TestProviders>
+    <SaveTimelineButton {..._props} />
+  </TestProviders>
+);
+
 describe('SaveTimelineButton', () => {
-  const props = {
-    initialFocus: 'title' as const,
-    timelineId: 'timeline-1',
-    toolTip: 'tooltip message',
-  };
-  test('Show tooltip', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(true);
+  beforeEach(() => {
+    jest.clearAllMocks();
   });
-  test('Hide tooltip', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
-    component.find('[data-test-subj="save-timeline-button-icon"]').first().simulate('click');
-    expect(component.find('[data-test-subj="save-timeline-btn-tooltip"]').exists()).toEqual(false);
+
+  // skipping this test because popover is not getting visible by RTL gestures.
+  //
+  // Raised a bug with eui team: https://github.com/elastic/eui/issues/6065
+  xit('Show tooltip', async () => {
+    const { container } = render(<TestSaveTimelineButton {...props} />);
+    const saveTimelineIcon = container.querySelectorAll(
+      '[data-test-subj="save-timeline-button-icon"]'
+    )[0];
+
+    fireEvent.mouseOver(saveTimelineIcon);
+
+    jest.runAllTimers();
+
+    await waitFor(() => {
+      expect(screen.getByRole('tooltip')).toBeVisible();
+    });
   });
-  test('should show a button with pencil icon', () => {
-    const component = mount(
-      <TestProviders>
-        <SaveTimelineButton {...props} />
-      </TestProviders>
-    );
+
+  it('Hide tooltip', async () => {
+    const { container } = render(<TestSaveTimelineButton {...props} />);
+
+    const saveTimelineIcon = container.querySelectorAll(
+      '[data-test-subj="save-timeline-button-icon"]'
+    )[0];
+
+    fireEvent.click(saveTimelineIcon);
+
+    await waitFor(() => {
+      expect(
+        container.querySelector('[data-test-subj="save-timeline-btn-tooltip"]')
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  it('should show a button with pencil icon', () => {
+    const { container } = render(<TestSaveTimelineButton {...props} />);
     expect(
-      component.find('[data-test-subj="save-timeline-button-icon"]').first().prop('iconType')
-    ).toEqual('pencil');
+      container.querySelectorAll('[data-test-subj="save-timeline-button-icon"]')[0].firstChild
+    ).toHaveAttribute('data-euiicon-type', 'pencil');
   });
-  test('should not show a modal when showOverlay equals false', () => {
-    const component = mount(
+
+  it('should have edit timeline btn disabled with tooltip if user does not have write access', () => {
```

Comment:
👍🏾 I would remove the tooltip part though since this doesn't test the tooltip. Maybe add a test that tries clicking and confirming the modal doesn't show as well?

Reply:
This particular tests that tooltip is visible when user does not have CRUD Access. But based on your suggestion, I have added one more tests to make sure that modal is now showing when user does not have CRUD Access.